pkg/vm/vm.go

```diff
@@ -950,6 +955,36 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 
 		v.estack.PushItem(stackitem.NewBigInteger(new(big.Int).Sqrt(a)))
 
+	case opcode.MODMUL:
+		modules := v.estack.Pop().BigInt()
+		if modules.Sign() == 0 {
+			panic("zero module")
+		}
+		x2 := v.estack.Pop().BigInt()
+		x1 := v.estack.Pop().BigInt()
+
+		v.estack.PushItem(stackitem.NewBigInteger((new(big.Int)).Mod(x1.Mul(x1, x2), modules)))
+
+	case opcode.MODPOW:
+		modules := v.estack.Pop().BigInt()
+		exponent := v.estack.Pop().BigInt()
+		base := v.estack.Pop().BigInt()
+		isInverse := exponent.Cmp(bigMinusOne) == 0
+		if isInverse {
+			if base.Cmp(bigZero) <= 0 {
+				panic("invalid base")
+			}
+			if modules.Cmp(bigTwo) < 0 {
+				panic("invalid module")
+			}
```

Comment:
What about dedicated `(big.Int).ModInverse` in this case?

Reply:
Do we need it? `Exp` handles this case.

Reply:
Not sure if we need it, but it is a bit faster and we branch in our code anyway.

Reply:
Fixed.